npm_and_yarn/spec/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater_spec.rb

```diff
@@ -209,6 +209,24 @@
       end
     end
 
+    context "when the package current-name is not defined in package.json" do
+      let(:files) { project_dependency_files("npm8/current_name_is_missing") }
+
+      it "restores the packages name attribute" do
+        parsed_lockfile = JSON.parse(updated_npm_lock_content)
+        puts JSON.pretty_generate(parsed_lockfile)
+        expected_updated_npm_lock_content = fixture(
+          "updated_projects",
+          "npm8",
+          "current_name_is_missing",
+          "package-lock.json"
+        )
+        expected_parsed_lockfile = JSON.parse(expected_updated_npm_lock_content)
+
+        expect(parsed_lockfile).to eq(expected_parsed_lockfile), "Differences found"
```

Comment:
@abdulapopoola , @JamieMagee 
Since we don't have name in package-lock.json -> packages -> "" -> name , I didn't use following check. `expect(parsed_lockfile.dig("packages", "", "name")).to eq("project-name")`. So I am not sure do we expect to add the names in `package-lock.json -> packages -> ""` json section which is equivalent to current package.json.